packages/wallet/src/utils/transaction-hash.ts

```diff
@@ -0,0 +1,14 @@
+export const getTransactionHash = (hash: unknown): string => {
+  if (typeof hash === 'string') {
+    return hash
+  }
+  if (typeof hash === 'object' && hash !== null) {
+    if ('result' in hash && typeof hash.result === 'string') {
+      return hash.result
+    }
+    if ('txid' in hash && typeof hash.txid === 'string') {
+      return hash.txid
+    }
+  }
+  return String(hash)
```

Comment:
Currently defaults to String(hash) if unknown. Could be worth logging a warning when format is not recognized, to catch silent bugs more easily. Wdyt?